src/environment.cpp

```diff
@@ -131,21 +140,14 @@ void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 		// Setting is done
 		state->m_initialization_needed = false;
 	}
-
-	// The index of the first pointed thing that was not returned
-	// before. The last index which needs to be tested.
-	s16 lastIndex = state->m_iterator.m_last_index;
-	if (!state->m_found.empty()) {
-		lastIndex = state->m_iterator.getIndex(
-			floatToInt(state->m_found.top().intersection_point, BS));
-	}
```

Comment:
You can add this back, now that `state->m_found` does no longer contain `POINTABLE_NOT` pointed things. (As per change in `getSelectedActiveObjects`.)
(=> Only need to iterate through nodes till first object is reached.)

Reply:
fixed